lib/Sapi.php

```diff
@@ -67,7 +67,21 @@ static function sendResponse(ResponseInterface $response) {
             }
 
         }
-        file_put_contents('php://output', $response->getBody());
+
+        $body = $response->getBody();
+        if(is_null($body)) return;
+
+        $contentLength = $response->getHeader('Content-Length');
+        if($contentLength !== null) {
+            $output = fopen('php://output', 'wb');
+            if(is_resource($body) && get_resource_type($body) == 'stream') {
+                stream_copy_to_stream($body, $output, $contentLength);
```

Comment:
According to http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13, the content-length header is the length of the response body as sent on the socket. If we send a compressed response by php, it should be the compressed size then. If the http server decides to encode any further, then I think it must modify the content-length accordingly.